File: src/main/java/com/ptaf/ai/validation/RunnableFeatureResult.java
package com.ptaf.ai.validation;

import java.util.List;

public record RunnableFeatureResult(
        boolean runnable,
        List<String> blockingReasons,
        List<String> warnings,
        boolean parseSuccessful,
        boolean stepValidationPassed,
        boolean yamlValidationPassed,
        boolean allowedYamlPassed,
        double stepReusePercentage,
        int totalSteps,
        int matchedSteps,
        int unmatchedSteps,
        int yamlKeysUsed,
        int existingYamlKeys,
        int missingYamlKeys
) {
}

File: src/main/ptaf/ai/validation/runnable_feature_result.py
from __future__ import annotations

from dataclasses import dataclass


@dataclass(frozen=True)
class RunnableFeatureResult:
    runnable: bool
    blocking_reasons: list[str]
    warnings: list[str]
    parse_successful: bool
    step_validation_passed: bool
    yaml_validation_passed: bool
    allowed_yaml_passed: bool
    page_frame_context_passed: bool
    step_reuse_percentage: float
    total_steps: int
    matched_steps: int
    unmatched_steps: int
    yaml_keys_used: int
    existing_yaml_keys: int
    missing_yaml_keys: int
    frame_step_count: int
    page_step_count: int
